packages/use-sync-external-store/src/useSyncExternalStoreWithSelector.js

```diff
@@ -54,12 +54,44 @@ export function useSyncExternalStoreWithSelector<Snapshot, Selection>(
     let hasMemo = false;
     let memoizedSnapshot;
     let memoizedSelection: Selection;
+    let lastUsedProps: string[] = [];
+    let hasAccessed = false;
+    const accessedProps: string[] = [];
+
     const memoizedSelector = (nextSnapshot: Snapshot) => {
+      const getProxy = (): Snapshot => {
+        if (
```

Comment:
@markerikson you were curious for the case when a a primitive value is passed, this lines will handle the same.
For Primitive Values

1. No Proxy wrapping occurs
2. The primitive value is passed directly to the selector
3. The property access tracking system (with accessedProps) is bypassed entirely